src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -424,6 +433,22 @@ internal bool Remove(TKey key)
             }
 
             //----------------------------------------------------------------------------------------
+            // Removes all entries from the table.
+            //
+            // Precondition:
+            //      Must hold _lock.
+            //----------------------------------------------------------------------------------------
+            internal void Clear()
+            {
+                // Remove all handles (as in Remove, just by setting the hashcode and leaving
+                // actual removal up to the finalizer).
+                for (int i = 0; i < _firstFreeEntry; i++)
```

Comment:
The handles will be still active and keeping the dependent objects alive, until the next rehash happens. It may take a long time, or potentially forever. We should also set the primary to null or something. Was the same problem in the original CoreCLR implementation?